src/index.ts

```diff
@@ -123,6 +124,17 @@ export default (opts: PluginOptions = {}): Plugin => {
               }
               return null
             })
+
+            if (parsedProject?.tsconfig && parsedImports) {
+              parsedProject.tsconfig.compilerOptions = {
+                ...parsedProject.tsconfig.compilerOptions,
+                paths: {
+                  ...parsedProject.tsconfig.compilerOptions?.paths,
+                  ...parsedImports,
+                },
```

Comment:
One problem with this is that it assumes that `compilerOptions.baseUrl` (the path that `compilerOptions.paths` are resolved with) is always the root directory, which is not true. This assumption exists because the `imports` field expects to always be resolved with the root directory.

Another problem is that the `imports` field is allowed to point to other packages. It can point to paths defined in the package's `exports` field too.

See [the spec](https://github.com/jkrems/proposal-pkg-exports/) for more details.